Change default UTF8 ccsid to 1208 and remove ISO8859-1 workarounds + Updates to Git builds and to Git 2.44

IgorTodorovskiIBM opened the pull request:
With this change, Git on z/OS now aligns the file tag (CCSID) with the git working-tree-encoding. 
Previously, there was a special case for UTF-8 encoded files. These files were tagged as ISO8859-1 (CCSID 819) because z/OS Open Tools sets _BPXK_AUTOCVT=ON, which doesn't auto-convert files tagged with the UTF-8 tag (CCSID 1208). 
Consequently, with this change, the default tag for UTF-8 encoded files is now UTF-8 (or CCSID 1208).

To adjust the default tag for UTF-8, you can configure the git setting `core.utf8ccsid` to 819 using the following commands:

- `git config --global core.utf8ccsid 819` # Global setting, 819 represents the CCSID for the UTF8 file tag
- `git config core.utf8ccsid 819` # Local setting affecting the current repository

Alternatively, you can set the GIT_UTF8_CCSID environment variable:

- `export GIT_UTF8_CCSID=819` # Environment variable

mfsysprog commented:
If someone uses GIT_UTF8_CCSID=819, shouldn't this logic remain in convert.c for that case?
```
+#ifdef __MVS__
+  // Don't convert ISO8859-1 on z/OS
+  if (strcasecmp("ISO8859-1", enc) == 0)
+    return 0;
+#endif
```
Also, the logic in utf8.c.patch should also only be applyed if someone sets GIT_UTF8_CCSID=819 I guess?
Or does the Rocket port also always handle iso8859-1 as utf8?

IgorTodorovskiIBM commented:
> If someone uses GIT_UTF8_CCSID=819, shouldn't this logic remain in convert.c for that case?
> 
> ```
> +#ifdef __MVS__
> +  // Don't convert ISO8859-1 on z/OS
> +  if (strcasecmp("ISO8859-1", enc) == 0)
> +    return 0;
> +#endif
> ```
> 
> Also, the logic in utf8.c.patch should also only be applyed if someone sets GIT_UTF8_CCSID=819 I guess? Or does the Rocket port also always handle iso8859-1 as utf8?

That makes sense, Rocket does not special case 819 from my understanding.

IgorTodorovskiIBM commented:
Updates:
* Guarded ISO8859-1 relevant code with utf8_ccsid == 819
* Updated the README blurb
* Updated to build off of git repo rather than tarball
* Updated to 2.43.2
